peps/pep-0747.rst

```diff
@@ -196,458 +170,199 @@ annotation expressions as input using plain ``object``\ s which would
 
-Implicit TypeExpr Values
-''''''''''''''''''''''''
+   x3 = TypeForm('type(1)')  # Error: invalid type expression
 
-Historically static type checkers have only needed to recognize
-*type expressions* in contexts where a type expression was expected.
-Now *type expression objects* must also be recognized in contexts where a
-value expression is expected.
+At runtime the ``TypeForm(...)`` callable simply returns the value passed to it.
 
-Static type checkers already recognize **class objects** (``type[C]``):
+This explicit syntax serves two purposes. First, it documents the developer's
+intent to use the value as a type form object. Second, static type checkers
+validate that all rules for type expressions are followed::
 
--  As a value expression, ``C`` has type ``type[C]``,
-   for each of the following values of C:
+  x4 = type(int)  # No error, evaluates to "type[int]"
+  
+  x5 = TypeForm(type(int))  # Error: call not allowed in type expression
 
-   -  ``name`` (where ``name`` must refer to a valid in-scope class, type alias, or TypeVar)
-   -  ``name '[' ... ']'``
-   -  ``<type> '[' ... ']'``
 
-The following **unparameterized type expressions** can be recognized unambiguously:
+Assignability
+-------------
 
--  As a value expression, ``X`` has type ``TypeExpr[X]``,
-   for each of the following values of X:
+``TypeForm`` has a single type parameter, which is covariant. That means
+``TypeForm[B]`` is assignable to ``TypeForm[A]`` if ``B`` is assignable to
+``A``::
 
-   -  ``<Any>``
-   -  ``<Self>``
-   -  ``<LiteralString>``
-   -  ``<NoReturn>``
-   -  ``<Never>``
+  def get_type_form() -> TypeForm[int]: ...
 
-**None**: The type expression ``None`` (``NoneType``) is ambiguous with the value ``None``,
-so must use the explicit ``TypeExpr(...)`` syntax:
+  t1: TypeForm[int | str] = get_type_form()  # OK
+  t2: TypeForm[str] = get_type_form()  # Error
 
--  As a value expression, ``TypeExpr(None)`` has type ``TypeExpr[None]``.
--  As a value expression, ``None`` continues to have type ``None``.
+``type[T]`` is a subtype of ``TypeForm[T]``, which means that ``type[B]`` is
+assignable to ``TypeForm[A]`` if ``B`` is assignable to ``A``::
 
-The following **parameterized type expressions** can be recognized unambiguously:
+  def get_type() -> type[int]: ...
 
--  As a value expression, ``X`` has type ``TypeExpr[X]``,
-   for each of the following values of X:
+  t3: TypeForm[int | str] = get_type()  # OK
+  t4: TypeForm[str] = get_type()  # Error
 
-   -  ``<Literal> '[' ... ']'``
-   -  ``<Optional> '[' ... ']'``
-   -  ``<Union> '[' ... ']'``
-   -  ``<Callable> '[' ... ']'``
-   -  ``<tuple> '[' ... ']'``
-   -  ``<TypeGuard> '[' ... ']'``
-   -  ``<TypeIs> '[' ... ']'``
+``TypeForm`` is a subtype of ``object`` and is assumed to have all of the
+attributes and methods of ``object``.
 
-.. _recognizing_annotated:
+If a value with a ``TypeForm`` type is used within an ``typing.assert_type``
```

Comment:
This feels problematic to me; assert_type() on a single expression should succeed only with a single type (or its equivalents).